spec/ruby/language/block_spec.rb

```diff
@@ -40,8 +40,16 @@ def m(a) yield a end
       m([1, 2]) { |a=5, b, c, d| [a, b, c, d] }.should == [5, 1, 2, nil]
     end
 
-    it "assigns elements to required arguments when a keyword rest argument is present" do
-      m([1, 2]) { |a, **k| [a, k] }.should == [1, {}]
+    ruby_version_is "3.2" do
+      it "does not autosplat single argument to required arguments when a keyword rest argument is present" do
+        m([1, 2]) { |a, **k| [a, k] }.should == [[1, 2], {}]
+      end
+    end
+
+    ruby_version_is ''..."3.2" do
+      it "autosplats single argument to required arguments when a keyword rest argument is present" do
+        m([1, 2]) { |a, **k| [a, k] }.should == [1, {}]
+      end
```

Comment:
If it is considered a bug, shouldn’t it be a `ruby_bug` guard?

Reply:
@eregon can you advise?

Reply:
`ruby_bug` mostly makes sense when the change will be backported, but it will probably not be backported here (https://bugs.ruby-lang.org/issues/18633#note-2).
The main difference between `ruby_bug` and `ruby_version_is` is what should be the behavior of other Ruby implementations.
In this case I think it's probably better for other Ruby implementations to follow the 3.0 & 3.1 behavior (assuming the fix is not backported), and so `ruby_version_is` seems better for compatibility of these other Ruby implementations (as long as they target 3.0/3.1).
It's also not hard to follow this behavior (I did it in TruffleRuby), although it does like a weird exception.

So I think ruby_version_is + `# https://bugs.ruby-lang.org/issues/18633` above is the best here.